scripts/comfyui.py

```diff
@@ -1,71 +1,122 @@
-            """)
+            gr.HTML(value=self.get_iframes_html(is_img2img))
+
         with gr.Row():
-            refresh_button = gr.Button(value=f'{ui.refresh_symbol} Reload ComfyUI interface (client side)', elem_id='sd-comfyui-webui-refresh_button')
+            refresh_button = gr.Button(value=f'{ui.refresh_symbol} Reload ComfyUI interface (client side)', elem_id=self.elem_id('refresh_button'))
             refresh_button.click(
                 fn=None,
                 _js='reloadComfyuiIFrames'
             )
-        return queue_front, output_node_label
 
-    def show(self, is_img2img):
-        return scripts.AlwaysVisible
+        return queue_front,
+
+    def get_iframes_html(self, is_img2img: bool) -> str:
+        comfyui_client_url = settings.get_comfyui_client_url()
+
+        iframes = []
+        first_loop = True
+        for workflow_type_id in external_code.get_workflow_type_ids(self.get_xxx2img_str(is_img2img)):
+            html_classes = ['comfyui-embedded-widget']
+            if first_loop:
+                first_loop = False
+                html_classes.append('comfyui-embedded-widget-display')
+
+            iframes.append(f"""
+                <iframe
+                    src="{comfyui_client_url}"
+                    workflow_type_id="{workflow_type_id}"
+                    class="{' '.join(html_classes)}"
+                    style="width:100%; height:500px;">
+                </iframe>
+            """)
+
+        return f"""
+            <div class="comfyui_iframes">
+                {''.join(iframes)}
+            </div>
+        """
 
-    def process(self, p, *args):
-        if not getattr(shared.opts, 'comfyui_enabled', True):
+    def process(self, p, queue_front, **kwargs):
+        if not getattr(global_state, 'enabled', True):
             return
 
-    def postprocess_batch_list(self, p, pp, queue_front, output_node_label, **kwargs):
-        if not getattr(shared.opts, 'comfyui_enabled', True):
+        global_state.queue_front = queue_front
+        workflow_patcher.patch_processing(p)
+
+    def postprocess_batch_list(self, p, pp, *args, **kwargs):
+        if not getattr(global_state, 'enabled', True):
             return
         if getattr(shared.state, 'interrupted', False):
             return
         if len(pp.images) == 0:
             return
 
-        batch_results = ComfyuiNodeWidgetRequests.start_workflow_sync(
-            batch=pp.images,
-            workflow_type='postprocess',
-            is_img2img=self.is_img2img,
-            required_node_types=webui_postprocess_output.expected_node_types,
-            queue_front=queue_front,
+        batch_results = external_code.run_workflow(
+            workflow_type=default_workflow_types.postprocess_workflow_type,
+            tab=self.get_xxx2img_str(),
+            batch_input=pp.images,
         )

```

Comment:
do you think it would be a good idea to use our own API instead of bypassing it to run workflow types? I'm talking about the code in `external_code.py`